### ISSUE TITLE
Add ability to limit the displayed decimal points for properties in the viewer footer

### ISSUE THREAD

**Issue by niksirbi** [Napari/napari#7238]
## 🚀 Feature
I propose adding the ability to control the displayed decimal points for properties in the viewer footer. The most straightforward way would be to inherit from `padas.set_option("display.precision")`.

## Motivation

I'm working with a Points layer, with an associated properties DataFrame. When I hover over a given point in the napari viewer, these properties are displayed on the bottom, which is a great feature!

What's not so great is that some of my properties have many decimal points, resulting in unnecessarily long numbers displayed in the viewer (see the circled "confidence" values in the screenshot below).

![Screenshot 2024-09-02 at 17 29 20](https://github.com/user-attachments/assets/d2f62817-4d14-44a3-868e-8d8508057d75)

Controlling the precision via `pandas.set_option("display.precision",  3)` is ineffective. I could round the properties, but for various reasons I want to avoid that loss in precision, if possible.

## Pitch

Since properties are stored as a pandas DataFrame, I'd expect pandas display options to effect how these properties appear in the viewer.

I propose altering the code [here](https://github.com/napari/napari/blob/4ea8d4fffe20bc9e3e033949ca05b64ec6f3c540/napari/layers/points/points.py#L2502-L2509) such that the printed property values inherit the display precision from pandas.

## Alternatives

Instead of inheriting the display precision from pandas, this could also be handled with a napari setting. But if such a setting doesn't already exist, creating one just for this use-case seems a bit of an overkill.

## Additional context

This matter was initially discussed on [Zulip](https://napari.zulipchat.com/#narrow/stream/212875-general/topic/Limit.20displayed.20decimal.20points.20for.20properties.20in.20viewer) and I've opened this issue as prompted by @jni.

I'm happy to have a go at opening a draft PR, if people agree with the pitched solution.

One thing I currently don't know is how this is handled for other layer types, and whether the proposed solution should work across layer types (as opposed to only the Points layer).

**Comment by jni** [comment]
With a bit more thought, I agree with @brisvag's comment [on Zulip](https://napari.zulipchat.com/#narrow/stream/212875-general/topic/Limit.20displayed.20decimal.20points.20for.20properties.20in.20viewer/near/467006978) that we don't actually want to be bound to pandas specifically (as opposed to "some columnar data analysis library, could be pandas or polars or narwhals or dask...), so we probably instead want our own setting.

I also think display should be based on *significant* digits, not digits after the decimal. Python's [float formatting](https://docs.python.org/3/library/string.html#format-specification-mini-language) [has this facility](https://stackoverflow.com/questions/25780022/how-to-make-python-format-floats-with-certain-amount-of-significant-digits) with the `'g'` flag:

```
In [1]: x = 1.0654820104354

In [2]: f'{x:.4g}'
Out[2]: '1.065'

In [3]: y = 22.7821

In [4]: f'{y:.4g}'
Out[4]: '22.78'
```

I think 4 is a reasonable default number (though I was taught 3 at school 😂).

**Comment by psobolewskiPhD** [comment]
Thanks for opening the issue!
I think a setting for precision globally in napari could be worth it -- we already have a hard-code 3 decimals utility function used for intensity values in the status bar:
https://github.com/napari/napari/blob/a9e7375e339a2240f6cc6affd05eceefa2e843e9/napari/utils/status_messages.py#L8-L10

So one option would be to use that function and then also make the value a setting?

**Comment by brisvag** [comment]
Could we somehow just expose the formatting specifier itself? So one could just pass `.4g` or `.f2` or whatever they prefer, and it would Just Work™?

**Comment by jni** [comment]
> Could we somehow just expose the formatting specifier itself?

yes but: I think that's more flexibility at the cost of readability — many people will have no idea how to type a valid formatter. We could do a two stage dialog: "floating point formatting: n significant digits OR specify formatter:". And we should have a preview box that changes as you choose the option.

I'm vaguely inclined to only do the simple setting first though, because the double setting just adds complexity...

**Comment by Czaki** [comment]
> yes but: I think that's more flexibility at the cost of readability — many people will have no idea how to type a valid formatter. We could do a two stage dialog: "floating point formatting: n significant digits OR specify formatter:". And we should have a preview box that changes as you choose the option.

Maybe just a text field with validation logic, that will treat entering number as a number of digits, otherwise accept use it as format string?

**Comment by andy-sweet** [comment]
Agreed with a bunch of comments above.

I think the first step is to use `napari.utils.status_messages.status_format` in `Labels._get_properties` instead of the plain feature value. Then consider adding one setting to control the floating point precision with the standard Python f-string format specifiers.

**Comment by mambroset** [comment]
> Thanks for opening the issue! I think a setting for precision globally in napari could be worth it -- we already have a hard-code 3 decimals utility function used for intensity values in the status bar:
> 
> [napari/napari/utils/status_messages.py](https://github.com/napari/napari/blob/a9e7375e339a2240f6cc6affd05eceefa2e843e9/napari/utils/status_messages.py#L8-L10)
> 
> Lines 8 to 10 in [a9e7375](/napari/napari/commit/a9e7375e339a2240f6cc6affd05eceefa2e843e9)
>  def format_float(value): 
>      """Nice float formatting into strings.""" 
>      return f'{value:0.3g}' 
> 
> So one option would be to use that function and then also make the value a setting?

Hello! Any update on this? I ran into the same limitation while using napari to inspect post-registration spatial Jacobian determinant maps. The values have very small floating-point variations (e.g. in the range `[0.994, 1.004]`), and with the current `.3g` formatting, all values above `1.0` display as `1` in the status bar — making the hover readout useless for this use case.
I currently work around it by monkey-patching `napari.utils.status_messages.format_float` to use `.6g`, which works but is obviously fragile. A global precision setting would be very welcome :)
Thanks!

**Comment by brisvag** [comment]
No updates @mambroset, but I think the discussion above is still valid and PRs on this are very welcome! Would you like to take a stab at it? :)

**Comment by mambroset** [comment]
@brisvag okay, I can try! It'll be my first PR ever, I'll submit it shortly, hope it'll be okay!
